### ISSUE TITLE
debugger and gtk3 issue

### ISSUE THREAD

**Issue by markuman** [geany/geany-plugins#756]
It seems like the debugger plugins is not compatible with gtk3.

Therefore I compiled geany 1.33 with `./configure --disable-gtk3`  
Result:

```
Using GTK version                : 2.24.32
```

When I now run `./autogen.sh` for the plugins, it results in

```
geany-plugins-1.34

  Build Environment:
    Geany version:                1.33 (GTK3)
    Install prefix:               /usr/local
    Datadir:                      /usr/local/share/geany-plugins
    Libdir:                       /usr/local/lib/geany-plugins
    Docdir:                       /usr/local/share/doc/geany-plugins
    Plugins path:                 /usr/lib/geany

  Build Features:
    Unit tests:                   yes
    Static code checking:         yes
    Extra C compiler warnings:    yes
    Extra linker options:         yes

  Plugins:
    Addons:                       yes
    Autoclose:                    yes
    Automark:                     yes
    CodeNav:                      yes
    Commander:                    yes
    Debugger:                     no
...
```

So it detect my self-compiled geany-gtk2 as gtk3 and still won't enable the debugger?  
Any ideas?

**Comment by elextr** [comment]
Do you have the development version of all the dependencies installed?  eg libvte?

**Comment by markuman** [comment]
@elextr I guess yes, but I'm not 100% sure. I'm running arch linux.

`community/libvterm 0.1.git1.c4317a6-3` was missing (I'm not sure if it's the same libvte what you've mentioned), but it makes no difference. Debugger is  still flagged as `no`.

**Comment by markuman** [comment]
```
# pacman -Ss vte
extra/vte-common 0.52.2-1 [installed]
    Common files used by vte and vte3
extra/vte3 0.52.2-1 [installed]
    Virtual Terminal Emulator widget for use with GTK3
```

**Comment by elextr** [comment]
no, libvte, not libvterm and you need the development package to compile against, not just the part needed to run it.  Also given arch's bleeding edge approach it may not have a package for the gtk2 libvte.

**Comment by elextr** [comment]
Or you could try applying https://github.com/geany/geany-plugins/pull/645 and see if you can build GTK3 debugger.

**Comment by markuman** [comment]
Afaik, arch does not split packages between libs and just runtimes. anyway. I try vte-legacy, which is build against gtk2: https://aur.archlinux.org/packages/vte-legacy/  
but the result is the same.  

I try now #645

**Comment by markuman** [comment]
So, after rebuilding everything, the debugger plugin is compiling fine with pr/645 and gtk3. I will test now if it is also working as expected.

**Comment by markuman** [comment]
seems to work. thx @elextr 

![debugger](https://user-images.githubusercontent.com/3920157/41902518-edbee0b8-7933-11e8-9f1d-eb02de016e49.gif)

**Comment by codebrainz** [comment]
@markuman FWIW, you probably just needed to use `--with-geany-libdir=/usr/local/lib` as it looks like it was picking up a Geany installed in `/usr` instead of `/usr/local`.

**Comment by markuman** [comment]
@codebrainz nope, I removed the geany package before used my own compiled version.

**Comment by codebrainz** [comment]
@markuman Do you have a file called `/usr/lib/pkgconfig/geany.pc`?

**Comment by markuman** [comment]
@codebrainz yes. but it's not installed

```
[bergholz@xps13 geany-plugins]$ cat /usr/lib/pkgconfig/geany.pc
prefix=/usr
exec_prefix=/usr
libdir=${exec_prefix}/lib
includedir=${prefix}/include
datarootdir=${prefix}/share
datadir=${datarootdir}
localedir=${datarootdir}/locale

Name: Geany
Description: A fast and lightweight IDE using GTK+
Requires: gtk+-3.0 >= 3.0 glib-2.0 >= 2.32
Version: 1.33
Libs: -L${libdir} -lgeany
Cflags: -DGTK -I${includedir}/geany -I${includedir}/geany/tagmanager -I${includedir}/geany/scintilla
[bergholz@xps13 geany-plugins]$ pacman -Ss geany
community/geany 1.33.0-2
    Fast and lightweight IDE
community/geany-plugins 1.33-3
    Plugins for Geany
[bergholz@xps13 geany-plugins]$ 
```

**Comment by codebrainz** [comment]
@markuman that's the file Geany-Plugins uses to detect your Geany (unless you use `--with-geany-libdir` to point it elsewhere), so that's why it was detecting your GTK+3 build instead of the GTK+2 build you had installed in a lower priority directory.
